### ISSUE TITLE
Dark mode in new version of dataTables

### ISSUE THREAD

**Issue by stla** [rstudio/DT#1085]
@yihui 

The latest version of **dataTables** has a [dark mode]. Do you think we can enable it and how if we upgrade?

**Comment by yihui** [comment]
Yes, we can try to upgrade DataTables.

**Comment by stla** [comment]
Ok, maybe I'll do the upgrade this week.

But how to assign the class `dark`?

**Comment by yihui** [comment]
> But how to assign the class `dark`?

I don't think you can do it in **DT**. At least it's not straightforward. You'll have to use JavaScript as instructed at https://datatables.net/manual/styling/dark-mode

**Comment by stla** [comment]
Nothing works :-(

![issueDTdark](https://github.com/rstudio/DT/assets/4466543/abf08455-9979-4400-9b60-7e360a3e7ae1)

Should I download everything here: 

![DTdownload](https://github.com/rstudio/DT/assets/4466543/034e66f3-08b3-4177-828a-f15595da662a)

**Comment by stla** [comment]
Wait, I did something wrong. I restart.

**Comment by stla** [comment]
It works, but I don't see any darkness

![dt_not_dark](https://github.com/rstudio/DT/assets/4466543/e9a79e29-97da-4adc-8233-28c8ebced9e3)

**Comment by stla** [comment]
I did
```r
library(shiny)
library(bslib)
library(DT)

js <- '
$(document).ready(function() {
  $("html").attr("data-bs-theme", "dark");
});
'

ui <- page_sidebar(
  tags$head(
    tags$script(
      HTML(js)
    )
  ),
  DTOutput("dtable"),
  sidebar = sidebar()
)

server <- function(input, output, session) {
  
  output[["dtable"]] <- renderDT({
    datatable(iris)
  })
  
}

shinyApp(ui, server)
```

Not dark. Looks like a css is missing?

**Comment by stla** [comment]
Hmm when I select some rows they are darker than usual:

![dtdarkselect](https://github.com/rstudio/DT/assets/4466543/bce8c4db-f981-4453-b0e9-8456cb6ea915)

**Comment by stla** [comment]
I tried 

```
$("html").attr("data-bs-theme", "dark")
```

and ```$("html").addClass("dark")```

That' strange because I can see the `html.dark` in the inpector tools. I also tried different values of the `styling` argument of  `datatable`. I tried with and without  **bslib**.

**Comment by stla** [comment]
Now I get some errors. This is totally crazy.

**Comment by stla** [comment]
I managed to get something dark outside shiny, using Bootstrap 5-3-2. But this is too dark!

![dtedge](https://github.com/rstudio/DT/assets/4466543/f2f6adfb-8a5d-4bd2-93fb-bec5dc328dce)

**Comment by stla** [comment]
I abdandon.

**Comment by yihui** [comment]
Well, this looks complicated indeed, but I think at least you need to tell **DT** to use the Bootstrap theme (e.g., `datatable(style = 'bootstrap')`. I don't think the default theme will work with the dark mode.

**Comment by stla** [comment]
Yes, it was just a typo.

I've just managed to get it outside Shiny !!:

![dtdark](https://github.com/rstudio/DT/assets/4466543/f7fa8379-6a26-4cf8-bad9-a0e6687bbb84)

Beautiful! Except the paging.

**Comment by stla** [comment]
Note that I got this one with Bootstrap 15-3-2 (downloaded the files on the datatables website).

What should I download otherwise ? **bslib** only supports 15-2 I believe.

**Comment by stla** [comment]
No way with Bootstrap 4.

**Comment by stla** [comment]
I abandon again.

So to sum up you have to add Bootstrap 5-3-2 as a HTML dependency and it works outside Shiny (up to the ugly pagination).

**Comment by stla** [comment]
With `style = "bootstrap4` the pagination is nice (still outside Shiny):

![dtdarkbs4](https://github.com/rstudio/DT/assets/4466543/d975bec5-e35c-4085-b283-3c0222152166)

**Comment by stla** [comment]
WORKS IN SHINY !!!!!!!

The "trick" was to put Bootstrap-5.3.2 in the **www** folder.

![dtdarkshiny](https://github.com/rstudio/DT/assets/4466543/c120db1f-14aa-4ddd-b732-5d0aded00e93)

```r
library(shiny)
library(DT)

ui <- fluidPage(
  tags$head(
    tags$link(href="bootstrap-5-3-2/css/bootstrap.min.css", rel="stylesheet"),
    tags$script(src="bootstrap-5-3-2/js/bootstrap.min.js")
  ),
  DTOutput("dtable"),
)

server <- function(input, output, session) {
  
  output[["dtable"]] <- renderDT({
    dtable <- datatable(
      iris, style= "bootstrap4", 
      options = list(initComplete = JS(
        'function() {$("html").attr("data-bs-theme", "dark");}')
      ))
  })
  
}

shinyApp(ui, server)
```

**Comment by stla** [comment]
What do we do? Even if the dark mode is suspicious, we can upgrade DataTables. We are not obliged to use the dark mode. I do a PR or not?

**Comment by yihui** [comment]
You can try to upgrade: https://github.com/rstudio/DT/blob/main/tools/update_DT.R The tricky thing is always testing. Thanks!

**Comment by stla** [comment]
Ok, will do soon. And maybe the dark mode works with **bslib**, I don't remember whether I tried or not...
